.github/workflows/test.yml

```diff
@@ -18,46 +18,78 @@ jobs:
   # This workflow contains a single job called "build"
   build:
     # The type of runner that the job will run on
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        julia-version: ['1.6']
+        os: [ubuntu-latest]
+        algorithm: ["AFPRegressor", "AIFeynman", "BSRRegressor", "DSRRegressor", "EHCRegressor", "FFXRegressor", "FEATRegressor", "FE_AFPRegressor", "EPLEXRegressor", "GPGOMEARegressor", "gplearn", "ITEARegressor", "MRGPRegressor", "OperonRegressor", "PySRRegressor", "sembackpropgp"]
+      # Keep running even with some algorithms failing:
+      fail-fast: false
     defaults:
       run:
         shell: bash -l {0}
     # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
       - uses: actions/checkout@v2
-      # Caches the conda environment
-      # - name: Cache conda
-      #   uses: actions/cache@v1
-      #   env:
-      #     # Increase this value to reset cache if environment.yml has not changed
-      #     CACHE_NUMBER: 0
-      #   with:
-      #     path: ~/conda_pkgs_dir
-      #     key:
-      #       ${{ runner.os }}-conda-${{ env.CACHE_NUMBER }}-${{
-      #       hashFiles('etc/example-environment.yml') }}
-      # Sets up the conda environment
+      - name: Clean out unused parts of environment.yml
+        run: |
+          python clean_environment.py ${{ matrix.algorithm }}
+          mv clean_environment.yml environment.yml
+
+      - name: Cache conda
+        uses: actions/cache@v2
+        env:
+          # Increase this value to reset cache if etc/example-environment.yml has not changed
+          CACHE_NUMBER: 0
+        with:
+          path: ~/conda_pkgs_dir
+          key:
+            ${{ matrix.os }}-conda-${{ env.CACHE_NUMBER }}-${{
+            hashFiles('environment.yml') }}
+
       - name: Set up Conda Environment 
         uses: conda-incubator/setup-miniconda@v2
         with:
           activate-environment: srbench
           environment-file: environment.yml
           auto-activate-base: false
+          # Use mamba:
+          mamba-version: "*"
+          channels: conda-forge,defaults
+          channel-priority: strict
+          # For caching:
+          use-only-tar-bz2: true
+
       - run: |
           conda info
           conda list
           conda activate srbench
 
+      - name: Set up Julia
```

Comment:
not sure about algorithm-specific setups here, eventually should move to docker setup